components/files/file_list/file_list.py

```diff
@@ -81,16 +62,49 @@ def main():
 
     headers = headers_from_jwt_or_oauth(a_jwt=args.jwt, a_client_id=args.oauth_id, a_client_secret=args.oauth_secret, a_scope=args.oauth_scope, a_server_config=server)
 
-    filesystem_domain_list = query_files(a_server_config=server, a_site_id=args.site_id, a_page_limit=args.page_limit, a_start=args.start, a_domain="file_system", a_view="v_fs_files_by_folder", a_headers=headers)
-    operator_domain_list = query_files(a_server_config=server, a_site_id=args.site_id, a_page_limit=args.page_limit, a_start=args.start, a_domain="operator", a_view="v_op_files_by_operator", a_headers=headers)
-
-    logging.info ("Found {} files in the 'file_system' domain".format(len(filesystem_domain_list)))
-    for fi in filesystem_domain_list:
-        logging.info (json.dumps(fi, sort_keys=True, indent=4))
-    
-    logging.info ("Found {} files in the 'operator' domain".format(len(operator_domain_list)))
-    for fi in operator_domain_list:
-        logging.info (json.dumps(fi, sort_keys=True, indent=4))
+    for state in ["active", "archived"]:
+        
+        page_traits = MetadataPaginationTraits(a_page_size=args.page_limit, a_start=args.start)
+        filesystem_domain_list = []
+        more_data = True
+        while more_data:
+            params = {"archived":True} if state == "archived" else {}
+            params=page_traits.params(a_params=params)
+            logging.debug("Using parameters:{}".format(json.dumps(params)))
+            file_page = query_metadata_by_domain_view(a_server_config=server, a_site_id=args.site_id, a_domain="file_system", a_view="v_fs_files_by_folder", a_headers=headers, a_params=params)
+            more_data = page_traits.more_data(file_page)
+            logging.info ("Found {} {} items {}".format(len(file_page["items"]), state, "({})".format("unpaginated" if page_traits.page_number() == 1 else "last page") if not more_data else "(page {})".format(page_traits.page_number())))
+            filesystem_domain_list.append(file_page["items"])

```

Comment:
Appending to a list and processing later undermines the goal of pagination. We should loop through the items and log them out in an internal loop.

Reply:
We could have another page traits which accepts a lambda that it calls for each individual item. This would hide away all pagination details.

Reply:
Another way to hide away pagination details would be to implement a python iterator for the view. This would give the caller more control than using lambdas